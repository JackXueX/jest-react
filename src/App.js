import React, { Component } from 'react';

class App extends Component {
  render() {
    return (
      <div className="container" title='dell lee' data-test='container'>
        hello World!
      </div>
    )
  }
}

export default App;