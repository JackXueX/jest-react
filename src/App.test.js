import React from 'react';
import App from '../src/App';
import { shallow } from 'enzyme'

describe('component APP test', () => {
  it('测试hello world', () => {
    const wrapper = shallow(<App />);
    // console.log(wrapper.find('.app-container').length);
    // expect(wrapper.find('.app-container').length).toBe(1);
    // console.log(wrapper.find('.app-container').prop('title'));
    // expect(wrapper.find('.app-container').prop('title')).toBe('dell lee');

    // expect(wrapper.find('[data-test="container]').prop('title')).toBe('dell lee');

    // console.log(wrapper.debug());
    // 测试和代码逻辑解藕
    // expect(wrapper.find('[data-test="container"]').prop('title')).toBe('dell lee');

    // const container = wrapper.find('[data-test="container"]');

    // expect(container).toExist();
    // expect(container).toHaveProp('title', 'dell lee');

    // 对于敏感内容的修改
    expect(wrapper).toMatchSnapshot();

  })
})