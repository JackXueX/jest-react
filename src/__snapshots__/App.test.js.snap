// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component APP test 测试hello world 1`] = `
<div
  className="container"
  data-test="container"
  title="dell lee"
>
  hello World!
</div>
`;
